osquery/sql/sqlite_util.cpp

```diff
@@ -175,6 +176,17 @@ SQLInternal::SQLInternal(const std::string& query, bool use_cache) {
   dbc->useCache(use_cache);
   status_ = queryInternal(query, results_, dbc);
 
+  if (status_.ok()) {
+    TableColumns tableColumns;
+    status_ = getQueryColumnsInternal(query, tableColumns, dbc);
```

Comment:
getQueryColumnsInternal is not reliable way for extracting the types

Reply:
Agreed.  Let's continue to discuss this in https://github.com/facebook/osquery/issues/4799